core/src/main/scala/akka/persistence/cassandra/snapshot/CassandraSnapshotStore.scala

```diff
@@ -347,6 +294,46 @@ class CassandraSnapshotStore(cfg: Config)
         b
     }
 
+    def serialize(payload: Any)(implicit ec: ExecutionContext): Future[Serialized] =
+      try {
+        def serializeMeta(): Option[SerializedMeta] =
+          // meta data, if any
+          payload match {
+            case SnapshotWithMetaData(_, m) =>
+              val m2 = m.asInstanceOf[AnyRef]
+              val serializer = serialization.findSerializerFor(m2)
+              val serManifest = Serializers.manifestFor(serializer, m2)
+              val metaBuf = ByteBuffer.wrap(serialization.serialize(m2).get)
+              Some(SerializedMeta(metaBuf, serManifest, serializer.identifier))
+            case _ => None
+          }
+
+        val p: AnyRef = (payload match {
+          case SnapshotWithMetaData(snap, _) => snap // unwrap
+          case snap                          => snap
+        }).asInstanceOf[AnyRef]
+        val serializer = serialization.findSerializerFor(p)
+        val serManifest = Serializers.manifestFor(serializer, p)
+        serializer match {
+          case asyncSer: AsyncSerializer =>
+            Serialization.withTransportInformation(system.asInstanceOf[ExtendedActorSystem]) { () =>
+              asyncSer.toBinaryAsync(p).map { bytes =>
+                val serPayload = ByteBuffer.wrap(bytes)
+                Serialized(serPayload, serManifest, serializer.identifier, serializeMeta())
+              }
+            }
+          case _ =>
+            Future {
+              // Serialization.serialize adds transport info
+              val serPayload = ByteBuffer.wrap(serialization.serialize(p).get)
+              Serialized(serPayload, serManifest, serializer.identifier, serializeMeta())
+            }
+        }
+
+      } catch {
+        case NonFatal(e) => Future.failed(e)
+      }
+
```

Comment:
Looks ok but worried I missed something that breaks old metadata logic somehow so that it won't work with multi-dc, do we have some tests for that using C* to make sure no regression?

Reply:
I'll run the multi dc tests with this just to be sure